addons/mail/models/mail_shortcode.py

```diff
@@ -21,3 +21,31 @@ class MailShortcode(models.Model):
     description = fields.Char('Description')
     message_ids = fields.Many2one('mail.message', string="Messages", store=False)
     last_used = fields.Datetime('Last Used', help="Last time this shortcode was used")
+    group_ids = fields.Many2many("res.groups", string="Authorized Groups", groups="base.group_system,im_livechat.im_livechat_group_manager")
```

Comment:
Not sure if feasible, but Authorized Groups should be a Many2one ideally, as groups can be composed, only one is necessary here to reach any possible combination.